### PR TITLE
ddl, core: fix parser issue 602

### PR DESCRIPTION
**Pull request by songyzh**
### What problem does this PR solve?

- Issue Number: https://github.com/pingcap/parser/issues/602

Problem Summary:

- Both TableOptionUnion and TableOptionInsertMethod are ignored by TiDB. 
- Expected behavior is to report an error

### What is changed and how it works?

What's Changed:

- add new errors
- if any of these two options found, report an error instead of ignoring
- add tests

```sql
# TableOptionUnion
mysql> CREATE TABLE z (a INT) ENGINE = MERGE UNION = (x, y);
ERROR 8231 (HY000): CREATE/ALTER table with union option is not supported

mysql> ALTER TABLE x UNION = (x, y);
ERROR 8231 (HY000): CREATE/ALTER table with union option is not supported


# TableOptionInsertMethod
mysql> CREATE TABLE z (a INT) ENGINE = MERGE INSERT_METHOD=LAST;
ERROR 8232 (HY000): CREATE/ALTER table with insert method option is not supported

mysql> ALTER TABLE z ENGINE = MERGE INSERT_METHOD=LAST;
ERROR 8232 (HY000): CREATE/ALTER table with insert method option is not supported
```

How it Works:

- check these two options in preprocess phase and report error if any of them exists

### Related changes

- no

Tests 

- Unit test

Side effects

- table won't be created or altered now, instead, error would be reported

### Release note 

- when creating or altering table with table union option or table insert_method option, error would be reported instead of these two options being ignored

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -2284,6 +2284,32 @@ func (s *testDBSuite1) TestCreateTable(c *C) {
 	//tk.MustGetErrCode(failSQL, errno.ErrDuplicatedValueInType)
 	//_, err = tk.Exec("create table t_enum (a enum('B','b'));")
 	//c.Assert(err.Error(), Equals, "[types:1291]Column 'a' has duplicated value 'B' in ENUM")
+
+	// test for table option "union" not supported
+	tk.MustExec("use test")
+	tk.MustExec("CREATE TABLE x (a INT) ENGINE = MyISAM;")
+	tk.MustExec("CREATE TABLE y (a INT) ENGINE = MyISAM;")
+	tk.MustExec("INSERT INTO x VALUES (1);")
+	tk.MustExec("INSERT INTO y VALUES (2);")
```
Comment (zimulala) [inline ddl/db_test.go:2293]: Could the above two lines be removed? It seems unnecessary.

File: ddl/db_test.go [hunk 2]
```diff
@@ -2284,6 +2284,32 @@ func (s *testDBSuite1) TestCreateTable(c *C) {
...
+	tk.MustGetErrCode(failSQL, errno.ErrTableOptionUnionUnsupported)
+	tk.MustExec("drop table x;")
+	tk.MustExec("drop table y;")
+
+	// test for table option "insert method" not supported
+	tk.MustExec("use test")
+	tk.MustExec("CREATE TABLE x (a INT) ENGINE = MyISAM;")
+	tk.MustExec("CREATE TABLE y (a INT) ENGINE = MyISAM;")
+	tk.MustExec("INSERT INTO x VALUES (1);")
+	tk.MustExec("INSERT INTO y VALUES (2);")
```
Comment (zimulala) [inline ddl/db_test.go:2306]: Ditto.

File: planner/core/preprocess.go [hunk 3]
```diff
@@ -485,6 +485,16 @@ func (p *preprocessor) checkCreateTableGrammar(stmt *ast.CreateTableStmt) {
...
+	for _, option := range stmt.Options {
+		switch option.Tp {
+		case ast.TableOptionUnion:
+			p.err = ddl.ErrTableOptionUnionUnsupported
+			return
+		case ast.TableOptionInsertMethod:
+			p.err = ddl.ErrTableOptionInsertMethodUnsupported
+			return
+		}
+	}
```
Comment (zimulala) [inline planner/core/preprocess.go:497]: Is it possible to extract this code into a function？There are two places to use.

File: ddl/db_test.go (hunk 1 above)
Comment (songyzh) [reply ddl/db_test.go]: @zimulala 
sorry for late response
fixed in https://github.com/pingcap/tidb/pull/17995/commits/ddb75be7718b62f09829a189de688ce533bc712b
PTAL, thanks!

File: planner/core/preprocess.go (hunk 3 above)
Comment (songyzh) [reply planner/core/preprocess.go:490]: @zimulala 
fixed in https://github.com/pingcap/tidb/pull/17995/commits/3d7754c49c1e01443fea7012e8202f412a0c6bbf
PTAL, thanks!

**Pull request merged**